runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -323,36 +335,52 @@ public LayoutBuilder removeSequencerServers(Set<String> endpoints)
         return this;
     }
 
+    /**
+     * Remove an endpoint from a segment.
+     *
+     * @param endpoint The endpoint to remove
+     * @param layoutStripe the stripe to remove the endpoint from
+     * @param mode mode of this segment
+     * @throws LayoutModificationException
+     */
+    public void removeFromStripe(String endpoint, LayoutStripe layoutStripe,
+                                 Layout.ReplicationMode mode) throws LayoutModificationException {
+        if (layoutStripe.getLogServers().remove(endpoint)) {
+            if (layoutStripe.getLogServers().isEmpty()) {
+                throw new LayoutModificationException(
+                        "Attempting to remove all logunit in stripe. "
+                                + "No replicas available.");
+            }
+            if (mode == Layout.ReplicationMode.CHAIN_REPLICATION
+                    && layoutStripe.getLogServers().size() < CHAIN_REPLICATION_MIN_NUM_NODES) {
+                throw new LayoutModificationException(
+                        "Change will cause redundancy loss!");
+            } else if (mode == Layout.ReplicationMode.QUORUM_REPLICATION) {
+                throw new IllegalStateException("Unsupported operation!");
```

Comment:
probably should also be UnrecoverableCorfuError

Reply:
Done.